src/openvic-simulation/scripts/Condition.cpp

```diff
@@ -0,0 +1,659 @@
+#include "Condition.hpp"
+#include "openvic-simulation/GameManager.hpp"
+#include <vector>
+
+using namespace OpenVic;
+using namespace OpenVic::NodeTools;
+
+Condition::Condition(
+	std::string_view new_identifier, value_type_t new_value_type, scope_t new_scope,
+	scope_t new_scope_change, identifier_type_t new_key_identifier_type,
+	identifier_type_t new_value_identifier_type
+) : HasIdentifier { new_identifier }, value_type { new_value_type }, scope { new_scope },
+	scope_change { new_scope_change }, key_identifier_type { new_key_identifier_type },
+	value_identifier_type { new_value_identifier_type } {}
+
+ConditionNode::ConditionNode(
+	Condition const* new_condition, value_t&& new_value,
+	HasIdentifier const* new_condition_key_item,
+	HasIdentifier const* new_condition_value_item
+) : condition { new_condition }, condition_key_item { new_condition_key_item },
+	condition_value_item { new_condition_key_item }, value { std::move(new_value) } {}
+
+bool ConditionManager::add_condition(
+	std::string_view identifier, value_type_t value_type, scope_t scope, scope_t scope_change,
+	identifier_type_t key_identifier_type, identifier_type_t value_identifier_type
+) {
+	if (identifier.empty()) {
+		Logger::error("Invalid condition identifier - empty!");
+		return false;
+	}
+
+	if (value_type == 0) {
+		Logger::error("Condition ", identifier, " accepts no value type!");
+		return false;
+	}
+
+	if (value_type > 0b1111111) {
+		Logger::error("Condition ", identifier, " accepts an invalid value type!");
+		return false;
+	}
+
+	if (scope == 0) {
+		Logger::error("Condition ", identifier, " has no scope specified!");
+		return false;
+	}
+
+	if (scope > (scope_t::COUNTRY | scope_t::PROVINCE | scope_t::POP | scope_t::MAGIC)) {
+		Logger::error("Condition ", identifier, " has no scope specified!");
+		return false;
+	}
+
+	if ((value_identifier_type == identifier_type_t::NONE) && (value_type == value_type_t::IDENTIFIER)) {
+		Logger::error("Condition ", identifier, " has no identifier type!");
+		return false;
+	}
+
+	if ((value_identifier_type != identifier_type_t::NONE) && (value_type != value_type_t::IDENTIFIER)) {
+		Logger::warning("Condition ", identifier, " specified an identifier type, but doesn't have an identifier!");
+	}
```

Comment:
This is missing `return false;`, unless it's intentional and I'm missing something. Also you could check both this and the case above in one go using `(value_identifier_type == identifier_type_t::NONE) == (value_type == value_type_t::IDENTIFIER)`, at the cost of needing a unified error message.

Reply:
It's intentional here - I decided to leave this as a warning because it doesn't compromise anything.

It just means that the condition has an extra value that will go unused - nothing is missing.